Code/Engine/RendererVulkan/Device/DeviceVulkan.h

```diff
@@ -149,17 +202,36 @@ class EZ_RENDERERVULKAN_DLL ezGALDeviceVulkan : public ezGALDevice
   struct PerFrameData
   {
     ezGALFence* m_pFence = nullptr;
+
+    /// \brief These are all fences passed into submit calls. For some reason waiting for the fence of the last submit is not enough. At least I can't get it to work (neither semaphores nor barriers make it past the validation layer).
+    ezHybridArray<vk::Fence, 2> m_CommandBufferFences;
```

Comment:
Fences are optional on command submit so it should be enough to insert a fence for the last one in a frame.

Reply:
This will only work as long as the command buffers have semaphores to connect them. Once we have e.g. async compute blocks that are independent of the other render data this is no longer the case.